src/lib/hooks/useNotificationHandler.ts

```diff
@@ -418,3 +355,71 @@ export function useNotificationsHandler() {
     setShowLoggedOut,
   ])
 }
+
+export function storePayload(payload: NotificationPayload) {
+  storedPayload = payload
+}
+
+export function getNotificationPayload(
+  e: Notifications.Notification,
+): NotificationPayload | null {
+  if (
+    e.request.trigger == null ||
+    typeof e.request.trigger !== 'object' ||
+    !('type' in e.request.trigger) ||
+    e.request.trigger.type !== 'push'
+  ) {
+    return null
+  }
+
+  const payload = (
+    isIOS ? e.request.trigger.payload : e.request.content.data
+  ) as NotificationPayload
```

Comment:
So we're validating `trigger` existence, do we need to do so for `content.data`? Might be missing this in the docs but I don't see any mention of why we need to handle this differently between platforms. Might be worth a comment here if there's anything unique about this logic.